service/src/test/java/com/ge/predix/acs/privilege/management/dao/GraphResourceConsistencyTest.java

```diff
@@ -37,25 +51,98 @@ private void setupTitanGraph() throws InterruptedException, ExecutionException {
         GraphConfig.createSchemaElements(this.graph);
     }
 
-    @Test(/* threadPoolSize = 2, invocationCount = 2, */ dataProvider = "resourcesForTestConcurrent",
+    @Test(
+            /* threadPoolSize = 2, invocationCount = 2, */ dataProvider = "resourcesForTestConcurrent",
             successPercentage = 97)
-    public void testConcurrentWriteByZoneAndResourceIdentifier(final String resourceId) {
-        System.out
-                .println("thread_id: " + Long.toString(Thread.currentThread().getId()) + ", resource_id:" + resourceId);
+    public void testWriteByZoneAndResourceIdentifierConcurrent(final String resourceId)
+            throws PermanentLockingException {
         ZoneEntity z = new ZoneEntity();
         z.setName("z1");
         persistResourceToZoneAndAssert(z, resourceId, Collections.emptySet());
     }
 
+    @Test(threadPoolSize = 2, invocationCount = 100)
+    public void testDeleteConcurrent() {
+        ResourceEntity resourceEntity = persistResource1toZone1AndAssert();
+        assertThat(this.resourceRepository.findOne(resourceEntity.getId()), equalTo(resourceEntity));
+        this.resourceRepository.delete(resourceEntity.getId());
+        assertThat(this.resourceRepository.findOne(resourceEntity.getId()), nullValue());
+
+    }
+
+    @Test(threadPoolSize = 2, invocationCount = 100)
+    public void testFindAllConcurrent() throws PermanentLockingException {
```

Comment:
this comment applies to all tests:

This test is expected to fail for some invocations because of PermanentLockingException (unique index lock).  You could handle the exception, though you cannot put a threshold on how many calls can fail.

I would use successPercentage - its better to ensure that no more than 3% tests fail even if we dont track which specific failure.